src/Knp/Component/Pager/Paginator.php

```diff
@@ -38,7 +38,8 @@ class Paginator implements PaginatorInterface
         self::SORT_DIRECTION_PARAMETER_NAME => 'direction',
         self::FILTER_FIELD_PARAMETER_NAME => 'filterParam',
         self::FILTER_VALUE_PARAMETER_NAME => 'filterValue',
-        self::DISTINCT => true
+        self::DISTINCT => true,
+        self::PREVENT_PAGE_OUT_OF_RANGE => false,
```

Comment:
This is too basic. We should give 3 possibilities:
1) do nothing (BC)
2) fix with last page (like already proposed)
3) throw a custom exception

Reply:
I've pushed changes.